plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttachedToWall.java

```diff
@@ -0,0 +1,115 @@
+package com.denizenscript.denizen.objects.properties.material;
+
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.type.Gate;
+import org.bukkit.block.data.type.Lantern;
+
+public class MaterialAttachedToWall implements Property {
+
+    public static boolean describes(ObjectTag material) {
+        return material instanceof MaterialTag
+                && ((MaterialTag) material).hasModernData()
+                && (((MaterialTag) material).getModernData() instanceof Gate
+                || ((MaterialTag) material).getModernData() instanceof Lantern);
+    }
+
+    public static MaterialAttachedToWall getFrom(ObjectTag _material){
+        if (!describes(_material)) {
+            return null;
+        }
+        else {
+            return new MaterialAttachedToWall((MaterialTag) _material);
+        }
+    }
+
+    public static final String[] handledMechs = new String[] {
+            "attached_to_wall"
+    };
+
+    private MaterialAttachedToWall(MaterialTag _material) {
+        material = _material;
+    }
+
+    MaterialTag material;
+
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.attached_to_wall>
+        // @returns ElementTag(Boolean)
+        // @mechanism MaterialTag.attached_to_wall
+        // @group properties
+        // @description
+        // Returns whether this material is attached to a wall.
+        // For a lantern, this returns true if it is hanging from the ceiling.
+        // For a gate, this returns whether it is lowered to attach to a wall block.
+        // -->
+        PropertyParser.<MaterialAttachedToWall, ElementTag>registerStaticTag(ElementTag.class, "attached_to_wall", (attribute, material) -> {
+            if (material.isGate()) {
+                return new ElementTag(material.getGate().isInWall());
+            }
+            else if (material.isLantern()) {
+                return new ElementTag(material.getLantern().isHanging());
+            }
+            else { // Unreachable
+                return null;
+            }
+        });
+    }
+
+    public boolean isGate() { return material.getModernData() instanceof Gate; }

```

Comment:
... wat? Where are the newlines

Reply:
I looked at several material properties files and some were formatted like this, some had newlines. Wasn't sure which to do so I just picked randomly, I'll fix.

Reply:
None are formatted like that. You might have your IDE configured to visually fold small methods, but they're not actually folded like that in source.